modules/collections/server/controllers/collections.controller.js

```diff
@@ -198,49 +144,43 @@ module.exports = DBModel.extend({
     return this.addDocument(collectionId, documentId, "other");
   },
 
-  addDocument: function(collectionId, documentId, targetDocType) {
+  addDocument: function(collectionId, documentId, docType) {
     var self = this;
 
+    // find the target collection in the database
     return this.findById(collectionId).then(function(collection) {
       if (!collection) {
         return;
       }
+
       // hold a reference to the doc lists, so we can flex the function based on the type of document being processed
       var docLists = {
         main : collection.mainDocuments,
         other : collection.otherDocuments
       };
-      var watchDocListType = targetDocType === "main" ? "other" : "main";
-
-      // do not re-add the same document
-      if (!_.find(docLists[targetDocType], function(cd) { return cd.document._id.equals(documentId); })) {
 
-        // remove from the other document list, if it is there
-        var inWatchDocList = _.filter(docLists[watchDocListType], function(cd) { return cd.document._id.equals(documentId); });
-        _.each(inWatchDocList, function(otherDoc){
-          self.removeDocument(collectionId, otherDoc.document._id, watchDocListType);
-        })
+      // find the database record for the document we are adding
+      var Document = new DocumentClass(self.opts);
+      return Document.findById(documentId).then(function(document) {
```

Comment:
This got cleaned up to support the new document drag&drop reordering, and was simplified in the process.